### PR TITLE
Added distance between nodes

### PR DESCRIPTION
**Pull request by marc7s**

### CODE REVIEW

File: TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/LaneNode.cs [hunk 1]
```diff
@@ -9,21 +9,22 @@ public class LaneNode : Node<LaneNode>
...
+        public LaneNode(Vector3 position, Quaternion rotation, RoadNode roadNode, float distanceToNextNode) : this(position, rotation, roadNode, null, null, distanceToNextNode){}
         
         /// <summary>Creates a new lane node</summary>
         /// <param name="position">The position of the node</param>
         /// <param name="rotation">The rotation of the node</param>
         /// <param name="roadNode">The road node this lane node relates to</param>
         /// <param name="prev">The previous lane node. Pass `null` if there is no previous</param>
         /// <param name="next">The next lane node. Pass `null` if there is no next</param>
-        public LaneNode(Vector3 position, Quaternion rotation, RoadNode roadNode, LaneNode prev, LaneNode next)
+        public LaneNode(Vector3 position, Quaternion rotation, RoadNode roadNode, LaneNode prev, LaneNode next, float distanceToNextNode)
```
Comment (Marre-B) [inline TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/LaneNode.cs:20]: The new variable "distanceToNextNode" seems to be used in a backwards way when building the list in Lane.cs (see other comment). Maybe rename to "distanceToPreviousNode" ?

File: TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/Lane.cs [hunk 2]
```diff
@@ -37,7 +37,7 @@ public Lane(Road road, RoadNode roadNode, LaneType type, VertexPath path)
             this._path = path;
 
             // Create the start node for the lane
-            _start = new LaneNode(path.GetPoint(0), GetNodeRotation(0, _road.EndOfPathInstruction), roadNode);
+            _start = new LaneNode(path.GetPoint(0), GetNodeRotation(0, _road.EndOfPathInstruction), roadNode, 0);
```
Comment (Marre-B) [inline TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/Lane.cs:40]: Here "distanceToNextNode" is set to 0?

File: TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/Lane.cs [hunk 3]
```diff
@@ -51,7 +51,7 @@ public Lane(Road road, RoadNode roadNode, LaneType type, VertexPath path)
             {
                 // Update the prev and create a new current node
                 prev = curr;
-                curr = new LaneNode(path.GetPoint(i), GetNodeRotation(path.cumulativeLengthAtEachVertex[i], _road.EndOfPathInstruction), currRoadNode, prev, null);
+                curr = new LaneNode(path.GetPoint(i), GetNodeRotation(path.cumulativeLengthAtEachVertex[i], _road.EndOfPathInstruction), currRoadNode, prev, null, path.DistanceBetweenPoints(i - 1, i));
```
Comment (Marre-B) [inline TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/Lane.cs:54]: Seems like "distanceToNextNode" is actually being set to the distance to the previous node.

File: TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/LaneNode.cs (hunk 1 above)
Comment (marc7s) [reply TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/LaneNode.cs:20]: I can switch the logic to be distance to previous node instead, might make more sense

**Pull request merged**